mockgcp/pkg/storage/memory.go

```diff
@@ -125,7 +131,7 @@ func (s *typeStorage) Get(ctx context.Context, fqn string, dest proto.Message) e
 
 	existing, found := s.byKey[fqn]
 	if !found {
-		return apierrors.NewNotFound(schema.GroupResource{}, fqn)
```

Comment:
I like concentrating these concerns in the storage layer. However I think we still have 1 of the issues I originally brought up here. Why return an explicit code.NotFound error for Get() when the key is not found but not for Update() or Delete() when the key is not found (lines 98 and 111). Why not codes.AlreadyExists for Create() at line 74?

Reply:
Ah, because I want to save tasty PR treats for tomorrow/Monday :-)  But now you've gone and ruined the surprise...